vcpkg/troubleshoot/asset-caching.md

```diff
@@ -0,0 +1,122 @@
+caching errors if the download URL does not match your asset cache's URL. If
+you added `x-block-origin`, failing to restore an asset from one of the caches
+fails with an error message:
+
+```Console
+error: Failed to download from mirror set
+```
+
+## <a name="cache-failure"></a> Artifacts aren't being uploaded or restored from my asset cache
+
+The `x-azurl` asset cache backend provided by vcpkg, is designed to work with
+Azure Storage Containers, it may work with other storage services that accept
+PUT requests with simple token authentication.
+
+The configuration has the following format:
+
+`x-azurl,<url>,<sas>[,<rw>]`
+
+* `<url>`: The container's base URL
+* `<sas>`: A Shared Access Signature (SAS) token if working with Azure Storage
+  Containers; or an authentication request parameter if working with other providers.
+* `<rw>`: (Optional) read/write permission configuration
+
+> !NOTE
+> While designed to work with Azure Storage Containers. The `x-azurl` backend
+> can be used for storage services that accept requests in the form of
+> `<url>?<sas>`.
+
+For example, `x-azurl,https://contoso.com,token=TOKEN_VALUE,readwrite` results in
+a request in the form of `https://contoso.com?token=TOKEN_VALUE`.
+
+If your asset cache provider requires no authorization, the `<sas>` parameter can
+be left empty. For example, `x-azurl,https://contoso.com,,readwrite`.
+
+### Cause 1: The URL is not correctly formatted
+
+When setting up your asset cache, make sure that you include `https://` as part of
+the URL.
+
+Steps to resolve:
+
+1 - Verify that the configured URL is correct.
+
+### Cause 2: The authorization token is not correctly formatted
+
+vcpkg uses the `<sas>` parameter to authenticate to your asset cache provider.
+There are a variety of reasons why the authorization may not succeed.
+
+If you're using an Azure Storage Container, the `<sas>` parameter should contain
+only the SAS token without any other request parameter formatting. For example:
+
+`x-azurl,https://mystorageaccount.blob.core.windows.net/mystoragecontainer,sasvaluehere,readwrite`
+
+Read the documentation on generating [SAS
+tokens](/azure/storage/blobs/blob-containers-portal#generate-a-shared-access-signature)
+for more information. Make sure that the generated token's read and write
+permissions match your required use case.
+
+If you're using a different provider, you may need to format the request
+parameters correctly. For example, prefixing a parameter name before the token's
+value.
+
+`x-azurl,https://contoso.com,authorization=tokenvaluehere,readwrite`.
+
+Steps to resolve:
+
+1 - Verify that the authorization token is not expired
+2 - Verify that the authorization token has the correct permissions to your container
+3 - Verify that the authorization token's value is correct
+4 - Verify that the authorization token's format matches the expected format by
+your provider.
+
+## I get a "Failed to download from mirror set" error message
```

Comment:
I'm not 100% sure, but I think you can get this error message if there is a network issue or the mirror is down for some reason. Maybe add that as a set to resolve?